lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4818,6 +4818,11 @@ namespace Js
 #endif
         }
 
+        if (result == FALSE && (propertyOperationFlags & PropertyOperation_StrictMode))
+        {
+            JavascriptError::ThrowCantDelete(propertyOperationFlags, scriptContext, TaggedInt::ToString(index, scriptContext)->GetString());
```

Comment:
`index` here doesn't have to be a `TaggedInt`. Won't this hit an assert if it's not? I think it would be better to use `GetPropertyDisplayNameForError(index, scriptContext)` instead.

Reply:
What would be a test case for non-integer index property? (I've changed to use GetPropertyDisplayNameForError but I'm curious :))

---
In reply to: [172680947](https://github.com/Microsoft/ChakraCore/pull/4786#discussion_r172680947) [](ancestors = 172680947)

Reply:
I think this hits the path we're talking about:

```js
function foo() {
  'use strict';
  let a = new Uint8Array(16);
  let sym = Symbol();
  Object.defineProperty(a, sym, {
    configurable: false,
    writable: false,
    enumerable: false
  });
  delete a[sym];
}
foo();
```

Reply:
May be worth adding a test case for this. 👍

Reply:
Yes, TaggedInt with this snippet asserts:
ASSERTION 11560: (taggedint.inl, line 152) Ensure var is actually a 'TaggedInt' Failure: (Is(aValue))
FATAL ERROR: ch.exe failed due to exception code c0000420

and GetPropertyDisplayNameForError produces correct exception:
TypeError: Calling delete on 'Symbol()' is not allowed in strict mode

Thanks!

---
In reply to: [172688314](https://github.com/Microsoft/ChakraCore/pull/4786#discussion_r172688314) [](ancestors = 172688314)

Reply:
Added test for deleting a non-configurable non-indexed property (though not a symbol, as it was hitting the same path/problem)

---
In reply to: [172688754](https://github.com/Microsoft/ChakraCore/pull/4786#discussion_r172688754) [](ancestors = 172688754)